### PR TITLE
Licensechecker

### PR DESCRIPTION
**Pull request by daveol**
This Pull Request adds a script that prints out files with UNKNOWN licenses and also fixes some files

### CODE REVIEW

**Comment by mcallegari** [comment]
Adding licenses to .ui files is not a good idea for 2 reasons:
1- the Qt Company itself doesn't do it on released examples. See this for instance:
http://code.qt.io/cgit/qt/qtbase.git/tree/examples/widgets/painting/fontsampler/mainwindowbase.ui

2- UI files are normally edited with Qt Creator, which will probably delete the license header once we save the file

**Comment by daveol** [comment]
Well I would like there to be a license header in the ui files, the way Qtractor does it seems like a good idea

**Comment by mcallegari** [comment]
Alright. Is there a way to automate the change on each .ui file ?
I don't really have the time to do it manually or write a script myself :cry:

**Comment by daveol** [comment]
I'll just do it myself :smile:

File: check-licenses.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+##config
+#file extentions
+FILE_EXT[0]="cpp"
+FILE_EXT[1]="c"
+FILE_EXT[2]="h"
+FILE_EXT[3]="js"
+FILE_EXT[4]="qml"
+FILE_EXT[5]="ui"
+#FILE_EXT[6]="xml"
```
Comment (daveol) [inline check-licenses.sh]: We track files with the following extensions:
- cpp
- c
- h
- js
- qml
- ui

Are there any file extensions that I have missed? And are you happy with the ones I chose?

**Comment by daveol** [comment]
I think this is ready. Thoughts @mcallegari, @plugz  ?

**Comment by mcallegari** [comment]
Looks OK to me. I need to double check if it makes sense to add a license header to the Android manifest, but in case I can remove it later.

Thanks !

**Comment by peternewman** [comment]
Would it be worth adding this to Travis CI @daveol ?

**Pull request merged**